### DISCUSSION TITLE
Spacegroup info in qmof.json file for DFT relaxed structures.

### DISCUSSION THREAD

**Discussion by srinidhimula** [arosen93/QMOF#26] [General]
Hi,
I would like to filter based on spacegroup and get info of the relaxed MOF structures from the qmof.json file. Later I would like to use these relaxed structures for property calculations of my interest. Is it possible to get the spacegroup information from the current files?

Thanks
Srinidhi

**Comment by Andrew-S-Rosen** [comment]
Hi Srinidhi,

Certainly, that's possible! This can be done using Pymatgen's interface to spglib via the `pymatgen.symmetry.analyzer` module described [here](https://pymatgen.org/pymatgen.symmetry.analyzer.html). I've written up an example code for you below.

```python
from ase.io import read
import numpy as np
from pymatgen.io.ase import AseAtomsAdaptor
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer

adaptor = AseAtomsAdaptor()
mofs = [adaptor.get_structure(mof) for mof in read('qmof-geometries.xyz',index=':')]
refcodes = np.genfromtxt('qmof-refcodes.csv',delimiter=',',dtype=str)
spacegroups = {}
symprec = 0.1
for i, mof in enumerate(mofs):
	spacegroup = SpacegroupAnalyzer(mof,symprec=symprec).get_space_group_symbol()
	print('Spacegroup for '+refcodes[i]+' = '+spacegroup)
	spacegroups[refcodes[i]] = spacegroup
```

Because the DFT relaxations were done without symmetry constraints (and in some cases, such as for the hypothetical MOFs, the starting structures were not high symmetry), you may need to tweak `symprec` a bit since the relaxed structures will not be perfectly symmetric.

**Reply by srinidhimula** [reply]
Hi Andrew,
Thank you for the example code. I can get the spacegroup information now. Thanks for the help!

**Reply by Andrew-S-Rosen** [reply]
Glad to hear it! In v9+, I will plan to include the spacegroup information by default. Thanks for bringing this up!
